doozer/doozerlib/backend/rebaser.py

```diff
@@ -0,0 +1,1549 @@
+            if 'empty.repo' not in line:
+                if line.endswith('\n'):
+                    line = line[0:-1]  # remove trailing newline, if exists
+                filtered_content.append(line)
+
+        df_lines = filtered_content
+
+        # ART-8476 assert rhel version equivalence
+        if self.should_match_upstream:
+            el_version = metadata.branch_el_target()
+            df_lines.extend([
+                '',
+                '# RHEL version in final image must match the one in ART\'s config',
+                f'RUN source /etc/os-release && [ "$PLATFORM_ID" == platform:el{el_version} ]'
+            ])
+
+        df_content = "\n".join(df_lines)
+
+        if release:
+            release_suffix = f'-{release}'
+        else:
+            release_suffix = ''
+
+        # Environment variables that will be injected into the Dockerfile
+        # unless content.set_build_variables=False
+        build_update_env_vars = {  # Set A
+            'OS_GIT_MAJOR': major_version,
+            'OS_GIT_MINOR': minor_version,
+            'OS_GIT_PATCH': patch_version,
+            'OS_GIT_VERSION': f'{major_version}.{minor_version}.{patch_version}{release_suffix}',
+            'OS_GIT_TREE_STATE': 'clean',
+            'SOURCE_GIT_TREE_STATE': 'clean',
+            'BUILD_VERSION': version,
+            'BUILD_RELEASE': release if release else '',
+        }
+
+        # Unlike update_env_vars (which can be disabled in metadata with content.set_build_variables=False),
+        # metadata_envs are always injected into doozer builds.
+        metadata_envs: Dict[str, str] = {
+            '__doozer_group': self._runtime.group,
+            '__doozer_key': metadata.distgit_key,
+            '__doozer_version': version,  # Useful when build variables are not being injected, but we still need "version" during the build.
+        }
+        if metadata.config.envs:
+            # Allow environment variables to be specified in the ART image metadata
+            metadata_envs.update(metadata.config.envs.primitive())
+
+        df_fileobj = self._update_yum_update_commands(metadata, force_yum_updates, io.StringIO(df_content))
+        with Path(dfp.dockerfile_path).open('w', encoding="utf-8") as df:
+            shutil.copyfileobj(df_fileobj, df)
+            df_fileobj.close()
+
+        self._update_environment_variables(metadata, source, df_path, build_update_envs=build_update_env_vars, metadata_envs=metadata_envs)
+
+        # Inject build repos for Konflux
+        self._add_build_repos(dfp)
+
+        self._reflow_labels(df_path)
+
+    def _add_build_repos(self, dfp: DockerfileParser):
+        # Populating the repo file needs to happen after every FROM before the original Dockerfile can invoke yum/dnf.
+        dfp.add_lines(
+            "\n# Start Konflux-specific steps",
+            "RUN mkdir -p /tmp/yum_temp; mv /etc/yum.repos.d/*.repo /tmp/yum_temp/ || true",
+            "COPY .oit/signed.repo /etc/yum.repos.d/",
```

Comment:
I think we want to change this to be signed/unsigned depending on the version 

https://github.com/openshift-eng/art-tools/blob/24fd4652d8ef4b803b28c21c3d0d1f9aa6d21340/doozer/doozerlib/osbs2_builder.py#L161

Reply:
This part was taken from original Ashwin's PR. Let's address this in a separate PR.